### ISSUE TITLE
The world state is divided by blocks; each new block representing a new world state.

### ISSUE THREAD

**Issue by aenhsaihan** [chronaeon/beigepaper#14]
I have a colleague who is refuting this statement in your paper.

His main point of contention was that the world state is a couple hundred GB in size while a block is just several MB.

Just wanted to raise the issue and get your point of view.

**Comment by morenoh149** [comment]
It depends on how you define "new world state".

If you define it as combining the previous state (the chain) plus a diff (the newest block).

Or if you define it as purely the diff (the latest block).

It's probably more important to agree on how it's all working rather than debate what should be considered a new world state.

**Comment by psionski** [comment]
Hi, @morenoh149 

I thought the world state and the chain are different things? For example the world state is "account A has 10 ETH, account B has 0 ETH", while the chain is something like "A sends 5 ETH to B" (so basically inputs to state transition functions, not state), and when nodes see this they update the (internal, implicit) world state?

The yellow paper says it like this:

> Ethereum, taken as a whole, can be viewed as a transaction-based state machine: we begin with a genesis state and incrementally execute transactions to morph it into some final state.

Then further down:

> The world state (state), is a mapping between addresses (160-bit identifiers) and account states (a data structure serialised as RLP, see Appendix B). Though not stored on the blockchain, it is assumed
that [...]

The way it's said in your paper, seems to suggest transactions actually contain world state (or fragments thereof). This is what I found confusing...

**Comment by morenoh149** [comment]
Ha this isn't my paper. I'm just a GitHub junkie.

The diagram seems to agree with @psionski that the evm world state is outside of the blockchain 

![](https://github.com/ethereumbook/ethereumbook/blob/develop/images/evm-architecture.png)

So my first comment was wrong.

https://github.com/ethereumbook/ethereumbook/blob/develop/13evm.asciidoc#ethereum-state
says
```
 The world state is a mapping of Ethereum addresses (160-bit values) to accounts. At the lower level, each Ethereum address represents an account comprising an ether balance, a nonce, the account’s storage, and the account’s program code. An externally owned account will always have no code and an empty storage.
```

**Comment by chronaeon** [comment]
@aenhsaihan yes, this is inaccurate. Thanks for reporting!

Thanks @morenoh149 and @psionski  for furthering the discussion and educating me!

Would it be possible to finish this sentence?

> The World State is divided [incremented?] by _________; each new ________ represents a new World State.

**Comment by morenoh149** [comment]
I'd try `The world state is a mapping of Ethereum addresses to accounts.` and skip the second part.

**Comment by chronaeon** [comment]
Fixed in 5fd62d57896372ff0e8334aa08b32585f92037ca
